### ISSUE TITLE
linter: spreading an AST node loses `loc`, so a plugin's report throws and disables every JS plugin for the file

### ISSUE THREAD

**Issue by marksnode** [oxc-project/oxc#26139]
### What version of Oxlint are you using?

1.80.0 (also reproduced on 1.79.0)

### What command did you run?

oxlint -c .oxlintrc.json fixture.js

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": [],
  "categories": {
    "correctness": "off"
  },
  "jsPlugins": [{"name": "spread-loc", "specifier": "./plugin.mjs"}],
  "rules": {
    "spread-loc/report": "error"
  }
}
```

### What happened?

AST nodes expose `loc` as an accessor on the prototype, so `{...node}` does not copy it. A JS plugin
that builds a helper object out of a node — a common shape when a rule needs to carry `getValue()`-style
helpers around — and then reports with a `loc` taken from that copy ends up passing
`loc: {start: undefined, end: undefined}` to `context.report`. oxlint rejects it with
`TypeError: loc.start must be an object`, and one rejected report disables **every** JS plugin for the
rest of that file, so unrelated findings disappear from the run.

`range`, `start` and `end` are own properties on the same node and do survive the spread — only `loc` is lost.

**Reproduction**.

`plugin.mjs`
```js
const rule = {
    create: context => ({
        VariableDeclarator(node) {
            const token = {...node, helper: () => 'x'}
            context.report({node, message: 'x', loc: {start: token.loc?.start, end: token.loc?.end}})
        },
    }),
}
export default {meta: {name: 'spread-loc'}, rules: {report: rule}}
```

`fixture.js`
```js
const a = 1
```

```
$ npx oxlint@1.80.0 -c .oxlintrc.json fixture.js
fixture.js: error: Error running JS plugin. File path: …/fixture.js TypeError: `loc.start` must be an object
    at report (…/oxlint/dist/lint.js:14057:64)
```

**Where the difference is.** A probe rule printing the node's own properties:

```js
console.error(`hasOwn loc=${Object.hasOwn(node, 'loc')} | 'loc' in node=${'loc' in node}`
    + ` | hasOwn range=${Object.hasOwn(node, 'range')} | spread: loc=${JSON.stringify({...node}.loc)}`)
```

oxlint 1.79.0 and 1.80.0:
```
hasOwn loc=false | 'loc' in node=true | hasOwn range=true | spread: loc=undefined range=[6,11]
```

eslint 9.39.2, same probe and fixture:
```
hasOwn loc=true | spread: loc={"start":{"line":1,"column":6},"end":{"line":1,"column":11}}
```

**Impact.** `eslint-plugin-rtl-friendly@0.5.1` (latest published) hits this on every finding: its
`format()` helper builds tokens with `{...node, getValue, getRaw}` and `report()` forwards
`token?.loc?.start`. Because the throw takes down the file's other JS plugins too, enabling that plugin
also silently disabled `@stylistic`, import sorting and our own rules on any file it reported on. We
currently ship a small adapter that drops an unusable `loc` from the descriptor and lets the report fall
back to the node, at the cost of a coarser position.
